components/input-field/examples/50-scenarios.tsx

```diff
@@ -0,0 +1,165 @@
+import { GEL } from '@westpac/core';
+import { InputField, Input, InputBefore, InputAfter } from '@westpac/input-field';
+import { Button } from '@westpac/button';
+import { Select, Textarea } from '@westpac/text-input';
+import {
+	TickIcon,
+	VisibilityIcon,
+	VisibilityOffIcon,
+	SearchIcon,
+	ClearIcon,
+	RefreshIcon,
+} from '@westpac/icon';
+import { useState, ChangeEvent } from 'react';
+
+function Example({ brand }: { brand: object | ((...args: unknown[]) => unknown) }) {
+	const [type, setType] = useState('password');
+	const [count, setCount] = useState(0);
+	const [value, setValue] = useState('');
+	const [charCountMessage, setCharCountMessage] = useState('250 remaining');
+
+	const onChangeTextAreaInput = (event: ChangeEvent<HTMLInputElement>) => {
+		const { value } = event.target;
+		setCharCountMessage(`${250 - value.length} remaining`);
+	};
+
+	// Multi state group (Inline field validation flow from Figma design)
+	const mockPostAbaNumberAPI = async (abaNumber: string) => {
+		const validRes: { status: keyof typeof addOnAfterMap; bank: string; errorMessage?: undefined } =
+			{
+				status: 'valid',
+				bank: 'Capital One, New Orleans, Louisiana',
+			};
+
+		const invalidRes: {
+			status: keyof typeof addOnAfterMap;
+			bank?: undefined;
+			errorMessage: string;
+		} = {
+			status: 'invalid',
+			errorMessage: 'Routing number not found',
+		};
+
+		// Wait 1000ms to demonstrate the loading state
+		await new Promise((r) => setTimeout(r, 1000));
+
+		// Only one valid number for demonstration purposes
+		return Promise.resolve(abaNumber === '647453' ? validRes : invalidRes);
+	};
+
+	const [abaNumber, setAbaNumber] = useState('');
+	const [status, setStatus] = useState<keyof typeof addOnAfterMap>('idle');
+	const [errorMessage, setErrorMessage] = useState('');
+	const [supportMessage, setSupportMessage] = useState('');
+
+	const onClickCheckBtn = async () => {
+		// Set loading status on click
+		setStatus('loading');
+
+		// Check the submitted number
+		const { status, errorMessage = '', bank = '' } = await mockPostAbaNumberAPI(abaNumber);
+
+		// Update the status based on the result from the API
+		setStatus(status);
+
+		// Show appropriate message to the user
+		setErrorMessage(errorMessage);
+		setSupportMessage(bank);
+	};
+
+	const addOnAfterMap = {
+		idle: (
+			<InputAfter>
+				<Button look="hero" onClick={onClickCheckBtn}>
+					Check
+				</Button>
+			</InputAfter>
+		),
+		loading: <InputAfter icon={RefreshIcon} />,
```

Comment:
Ahh sorry about this - it wasn't as polished as I thought it was, e.g. this should probably rotate and announce that it's loading but I think, for demo purposes of state changes we've conveyed the message to any consumers that want to pick it up. I'm happy to leave this as-is, if you are?

Reply:
Yea happy to leave as is